frontend/packages/knative-plugin/src/components/add/CatalogEventSourceForm.tsx

```diff
@@ -0,0 +1,132 @@
+import * as React from 'react';
+import * as _ from 'lodash';
+import { FormikProps, FormikValues } from 'formik';
+import { useTranslation } from 'react-i18next';
+import { Alert, AlertActionCloseButton } from '@patternfly/react-core';
+import { FormFooter, SyncedEditorField, YAMLEditorField, FlexForm } from '@console/shared';
+import { EditorType } from '@console/shared/src/components/synced-editor/editor-toggle';
+import { safeJSToYAML } from '@console/shared/src/utils/yaml';
+import { LoadingInline } from '@console/internal/components/utils';
+import CatalogEventSourceSection from './event-sources/CatalogEventSourceSection';
+import { EventSourceListData, SinkType, EventSourceSyncFormData } from './import-types';
+import { getCatalogEventSourceResource } from '../../utils/create-eventsources-utils';
+import { isDynamicEventSourceKind } from '../../utils/fetch-dynamic-eventsources-utils';
+
+interface OwnProps {
+  namespace: string;
+  eventSourceStatus: EventSourceListData | null;
+  eventSourceMetaDescription?: any;
+}
+
+const CatalogEventSourceForm: React.FC<FormikProps<FormikValues> & OwnProps> = ({
+  errors,
+  values,
+  handleSubmit,
+  handleReset,
+  setFieldValue,
+  status,
+  isSubmitting,
+  dirty,
+  namespace,
+  eventSourceStatus,
+  eventSourceMetaDescription,
+}) => {
+  const { t } = useTranslation();
+  const yamlEditor = <YAMLEditorField name="yamlData" onSave={handleSubmit} />;
+
+  const sanitizeToForm = (newFormData) => {
+    const specData = newFormData.spec;
+    const initialValue = {
+      project: {
+        name: newFormData.metadata.namespace || '',
+        displayName: '',
+        description: '',
+      },
+      application: {
+        initial: '',
+        name: '',
+        selectedKey: '',
+      },
+      name: newFormData.metadata.name,
+      apiVersion: newFormData.apiVersion,
+      sinkType: newFormData.spec?.sink?.ref ? SinkType.Resource : SinkType.Uri,
+      sink: {
+        apiVersion: newFormData.spec?.sink?.ref?.apiVersion,
+        kind: newFormData.spec?.sink?.ref?.kind,
+        name: newFormData.spec?.sink?.ref?.name,
```

Comment:
```suggestion
        apiVersion: specData?.sink?.ref?.apiVersion,
        kind: specData?.sink?.ref?.kind,
        name: specData?.sink?.ref?.name,
```

It will be even better if we destructure the props from  `specData?.sink?.ref?` in the top of the function .